bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/discovery/NetatmoDiscoveryService.java

```diff
@@ -71,40 +70,44 @@ public void startScan() {
-                                    ThingUID bridgeUID = createThing(station, accountUID);
-                                    station.getModules().values().stream()
-                                            .forEach(module -> createThing(module, bridgeUID));
-                                });
-                    }
+                WeatherApi weatherApi = localHandler.getRestManager(WeatherApi.class);
+                if (weatherApi != null) { // Search owned or favorite stations
+                    weatherApi.getFavoriteAndGuestStationsData().stream().forEach(station -> {
+                        if (!station.isReadOnly() || localHandler.getReadFriends()) {
+                            ThingUID stationUID = createThing(station, accountUID);
+                            station.getModules().values().stream().forEach(module -> createThing(module, stationUID));
+                        }
+                    });
                 }
                 HomeApi homeApi = localHandler.getRestManager(HomeApi.class);
-                if (homeApi != null) { // Search those who depend from a home
-                    homeApi.getHomesData(null, null).stream().filter(h -> !h.getFeatures().isEmpty()).forEach(home -> {
-                        ThingUID homeUID = createThing(home, accountUID);
-
-                        home.getKnownPersons().forEach(person -> createThing(person, homeUID));
-
-                        Map<String, ThingUID> bridgesUids = new HashMap<>();
-
-                        home.getRooms().values().stream().forEach(room -> {
-                            room.getModuleIds().stream().map(id -> home.getModules().get(id))
-                                    .map(m -> m != null ? m.getType().feature : FeatureArea.NONE)
-                                    .filter(f -> FeatureArea.ENERGY.equals(f)).findAny()
-                                    .ifPresent(f -> bridgesUids.put(room.getId(), createThing(room, homeUID)));
-                        });
-
-                        // Creating modules that have no bridge first
-                        home.getModules().values().stream().filter(module -> module.getBridge() == null)
-                                .forEach(device -> bridgesUids.put(device.getId(), createThing(device, homeUID)));
-                        // Then the others
-                        home.getModules().values().stream().filter(module -> module.getBridge() != null).forEach(
-                                device -> createThing(device, bridgesUids.getOrDefault(device.getBridge(), homeUID)));
-                    });
+                if (homeApi != null) { // Search those depending from a home that has modules + not only weather modules
+                    homeApi.getHomesData(null, null).stream()
+                            .filter(h -> !(h.getFeatures().isEmpty()
+                                    || h.getFeatures().contains(FeatureArea.WEATHER) && h.getFeatures().size() == 1))
```

Comment:
I think your new filter is not correct.
In my case, I have certainly several feature areas and it leads to the create of a thing for my weather station attached to the home thing.

Reply:
I want to avoid the creation of a home that only owns a station.
Normally, your station should have been created and discovered in the previous step, but maybe I made a mistake there.

The thing is, my own setup has three homes, one for weather, one for energy, one for security. Never found how to merge them. The weather home is pointless.

Reply:
Ok, I understand, your filter is for your case you have a home only for your weather station.
In my case, my home contains a weather station and a camera. In this case, the weather station should be ignored. This is an additional test to do in in the loop so that only security and energy things are created at this place.

Reply:
> Normally, your station should have been created and discovered in the previous step, but maybe I made a mistake there.

Exactly, that is why it should not be discovered a second time, attached to my home that time.

Reply:
> If its the case, we must find how to avoid creation of a second station.

A simple test on the thing type ? I do not remember exactly how look the returned data.

Reply:
I just checked the returned data, there is a "type" field for each "thing". Type "NAMain" should be ignored.

Reply:
I just checked the returned data, there is a "type" field for each "thing" in the home. Type "NAMain" should be ignored.

Reply:
Once the correct filter will be added, you can even remove your own filter which cover only a specific case that can be covered by the more general test.

Reply:
@clinique: here are the changes which works for me:
```
                                // Creating modules that have no bridge first
                                home.getModules().values().stream()
                                        .filter(module -> module.getBridge() == null
                                                && module.getType() != ModuleType.WEATHER_STATION)
                                        .forEach(device -> bridgesUids.put(device.getId(),
                                                createThing(device, homeUID)));

                                // Then the others
                                home.getModules().values().stream()
                                        .filter(module -> module.getBridge() != null
                                                && module.getType().getBridge() != ModuleType.WEATHER_STATION)
                                        .forEach(device -> createThing(device,
                                                bridgesUids.getOrDefault(device.getBridge(), homeUID)));
```